src/NzbDrone.Core/Download/Clients/Tribler/TriblerDownloadClient.cs

```diff
@@ -0,0 +1,324 @@
+            }
+
+            return new DownloadClientInfo
+            {
+                IsLocalhost = Settings.Host == "127.0.0.1" || Settings.Host == "localhost",
+                OutputRootFolders = new List<OsPath> { _remotePathMappingService.RemapRemoteToLocal(Settings.Host, new OsPath(destDir)) }
+            };
+        }
+
+        /**
+         * this basically checks if torrent is stopped because of seeding has finished
+         */
+        protected static bool HasReachedSeedLimit(Download torrent, GetTriblerSettingsResponse config)
+        {
+            if (config == null)
+            {
+                throw new ArgumentNullException(nameof(config));
+            }
+
+            if (torrent == null)
+            {
+                throw new ArgumentNullException(nameof(torrent));
+            }
+
+            // if download is still running then it's not finished.
+            if (torrent.Status != DownloadStatus.Stopped)
+            {
+                return false;
+            }
+
+            switch (config.Settings.DownloadDefaults.SeedingMode)
+            {
+                // if in ratio mode, wait for ratio to become larger than expeced. Tribler's DownloadStatus will switch from SEEDING to STOPPED
+                case DownloadDefaultsSeedingMode.Ratio:
+
+                    return torrent.Ratio.HasValue
+                        && torrent.Ratio >= config.Settings.DownloadDefaults.SeedingRatio;
+
+                case DownloadDefaultsSeedingMode.Time:
+                    var downloadStarted = DateTimeOffset.FromUnixTimeSeconds(torrent.TimeAdded.Value);
+                    var maxSeedingTime = TimeSpan.FromSeconds(config.Settings.DownloadDefaults.SeedingTime ?? 0);
+
+                    return torrent.TimeAdded.HasValue
+                        && downloadStarted.Add(maxSeedingTime) < DateTimeOffset.Now;
+
+                case DownloadDefaultsSeedingMode.Never:
+                    return true;
+
+                case DownloadDefaultsSeedingMode.Forever:
+                default:
+                    return false;
+            }
+        }
+
+        protected override string AddFromMagnetLink(RemoteEpisode remoteEpisode, string hash, string magnetLink)
+        {
+            var addDownloadRequestObject = new AddDownloadRequest
+            {
+                Destination = GetDownloadDirectory(),
+                Uri = magnetLink,
+                SafeSeeding = Settings.SafeSeeding,
+                AnonymityHops = Settings.AnonymityLevel
+            };
+
+            return _proxy.AddFromMagnetLink(Settings, addDownloadRequestObject);
+        }
+
+        protected override string AddFromTorrentFile(RemoteEpisode remoteEpisode, string hash, string filename, byte[] fileContent)
+        {
+            // tribler api currently do not support recieving a torrent file direcly.
+            throw new NotSupportedException("Tribler download client only support magnet links currently");
```

Comment:
Is there a GHI or feature request somewhere to add this support that can be referenced?

```suggestion
            throw new NotSupportedException("Tribler does not support torrent files");
```